### PR TITLE
[18.0][MIG] product_supplierinfo_archive: Migration to 18.0 #1771

### PR DESCRIPTION
**Pull request by pfranck**

### CODE REVIEW

**Review by sebalix — CHANGES_REQUESTED** [review]
You somehow squashed your migration commit into the pre-commit one

**Comment by pfranck** [comment]
Actually there was no fix -- the module work out of the box - that why we have only 1 commit.
unless you want me to create a new commit to change the version of the module

**Comment by pedrobaeza** [comment]
@sebalix I included the other this clarification on the migration commit because of the lack of pre-commit auto-fixes:

https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-18.0/_compare/3d012e10cc5c6f542cc1a47a4820a5cebb881cd2...a96cf3dc2c6e60f4b14eb45305cd1ed08501d77b

**Comment by pfranck** [comment]
So you don't expect people to bump the version 18.0 before running `git commit -m "[IMP] $module: pre-commit auto fixes"`

File: product_supplierinfo_archive/views/product_views.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    <record id="product_supplierinfo_form_view" model="ir.ui.view">
+        <field
+            name="name"
+        >product.supplierinfo.form.view - product_supplierinfo_archive</field>
+        <field name="model">product.supplierinfo</field>
+        <field name="inherit_id" ref="product.product_supplierinfo_form_view" />
+        <field name="arch" type="xml">
+            <field name="product_variant_count" position="after">
+                <field name="active" invisible="1" />
```
Comment (xaviedoanhduy) [inline product_supplierinfo_archive/views/product_views.xml:24]: > Missing fields used in domains and other attributes are added to the view as invisible automatically. If you encounter fields in views with invisible="True", they can probably be removed. Reference: https://github.com/odoo/odoo/pull/137031

File: product_supplierinfo_archive/views/product_views.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+<odoo>
+    <record id="product_supplierinfo_tree_view" model="ir.ui.view">
+        <field
+            name="name"
+        >product.supplierinfo.tree.view - product_supplierinfo_archive</field>
+        <field name="model">product.supplierinfo</field>
+        <field name="inherit_id" ref="product.product_supplierinfo_tree_view" />
+        <field name="arch" type="xml">
+            <field name="price" position="after">
+                <field name="active" invisible="1" />
```
Comment (xaviedoanhduy) [inline product_supplierinfo_archive/views/product_views.xml:11]: > Missing fields used in domains and other attributes are added to the view as invisible automatically. If you encounter fields in views with invisible="True", they can probably be removed. Reference: https://github.com/odoo/odoo/pull/137031

**Review by xaviedoanhduy — CHANGES_REQUESTED** [review]

**Comment by rousseldenis** [comment]
> So you don't expect people to bump the version 18.0 before running `git commit -m "[IMP] $module: pre-commit auto fixes"`

Nope, that's the flow. The pre-commit fixes should be applied with the configuration of the destination branch and commited with `--no-verify` configuration.

Then, the migration changes should be applied

File: product_supplierinfo_archive/views/product_views.xml (hunk 2 above)
Comment (xaviedoanhduy) [reply product_supplierinfo_archive/views/product_views.xml:11]: looks weird

image

Comment (sebalix) [reply product_supplierinfo_archive/views/product_views.xml:11]: @pfranck can you address this comment please? :pray:

**Comment by Kimkhoi3010** [comment]
Hi @pfranck, @pedrobaeza, @rousseldenis, @sebalix, I created https://github.com/OCA/product-attribute/pull/1825 to take over this PR.

**Comment by pedrobaeza** [comment]
Superseded by #1825

**Pull request closed without merging**